### PR TITLE
High Speed Tracer

### PR DESCRIPTION
**Pull request by ANormalTwig**
Basically a better high speed ranger since the high speed ranger is pretty useless with its limited values.

### CODE REVIEW

File: lua/entities/gmod_wire_hstracer/init.lua [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		self.tpsout = self.tps
+		self.tps = 0
+	end
+
+	self:SetOverlayText(string.format("High Speed Tracer\nTraces per second: %d", self.tpsout))
+end
+
+function ENT:Think()
+	local curTime = CurTime()
+	if curTime > self.tpstime then
```
Comment (thegrb93) [inline lua/entities/gmod_wire_hstracer/init.lua:114]: Use `self:NextThink(CurTime()+1) return true` instead

File: lua/entities/gmod_wire_hstracer/init.lua [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	self.Memory = {}
+
+	self.Memory[7] = MASK_SOLID
+	self.Memory[8] = COLLISION_GROUP_NONE
+
+	self:UpdateOverlay()
+end
+
+function ENT:Trace()
+	self.BaseClass.Think(self)
```
Comment (thegrb93) [inline lua/entities/gmod_wire_hstracer/init.lua:53]: This shouldn't call baseclass think, the Think function should

**Comment by thegrb93** [comment]
Why not update the wire ranger instead of making a new entity that does the same thing?

File: lua/entities/gmod_wire_hstracer/init.lua (hunk 2 above)
Comment (ANormalTwig) [reply lua/entities/gmod_wire_hstracer/init.lua:53]: ah, I must've put it in there by accident thinking it was the Think method

**Comment by ANormalTwig** [comment]
the high speed ranger or the trace gate?

**Comment by thegrb93** [comment]
High speed ranger, this is really similar to that. https://github.com/wiremod/wire-extras/blob/master/lua/entities/gmod_wire_hsranger/init.lua

**Comment by thegrb93** [comment]
Can we not just improve the highspeed range instead of creating this new thing?

**Comment by thegrb93** [comment]
I don't think adding new addresses would hurt it

**Comment by ANormalTwig** [comment]
Any ideas on how I should integrate all the current ones into the ranger then?

**Comment by thegrb93** [comment]
Just do whatever looks best without modifying the existing ones

**Comment by ANormalTwig** [comment]
christ, nevermind

**Pull request closed without merging**